### PR TITLE
newrelic-infrastructure-v3: explain rbac rules

### PR DESCRIPTION
**Pull request by roobre**
#### Is this a new chart
No

#### What this PR does / why we need it:
Adds a few comments explaining why we require permissions to access several endpoints/

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] Chart Version bumped
- [ ] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[mychartname]`)

### CODE REVIEW

File: charts/newrelic-infrastructure-v3/templates/clusterrole.yaml [hunk 1]
```diff
@@ -6,16 +6,23 @@ metadata:
...
     verbs: ["get", "list"]
   - apiGroups: [ "" ]
     resources:
+      # endpoints and services are used for target autodiscovery.
       - "endpoints"
       - "services"
+      # nodes is used to fetch node data.
       - "nodes"
     verbs: [ "get", "list", "watch" ]
   - nonResourceURLs: ["/metrics"]
```
Comment (roobre) [inline charts/newrelic-infrastructure-v3/templates/clusterrole.yaml:28]: `TODO`: Add explanation for this one.

Comment (paologallinaharbur) [reply charts/newrelic-infrastructure-v3/templates/clusterrole.yaml:28]: Should we merge it anyway or wait for a confirmation over this?

Comment (roobre) [reply charts/newrelic-infrastructure-v3/templates/clusterrole.yaml:28]: I would merge anyway. I'll mark this unread so I don't forget to rebase this :)

**Review by paologallinaharbur — APPROVED** [review]

**Comment by roobre** [comment]
After your PR with the deprecation I wonder if we should port this patch to the `nri-kubernetes` repo instead. WDYT @paologallinaharbur ?

**Comment by paologallinaharbur** [comment]
You are totally right, merging it here will result likely to have it lost since the chart has been already copied

**Comment by roobre** [comment]
Closing, to be tackled in https://github.com/newrelic/nri-kubernetes/pull/383 or similar.

**Pull request closed without merging**